pandas/tests/io/excel/test_readers.py

```diff
@@ -424,19 +423,11 @@ def test_reader_special_dtypes(self, request, read_ext):
 
         # if not coercing number, then int comes in as float
         float_expected = expected.copy()
-        float_expected["IntCol"] = float_expected["IntCol"].astype(float)
         float_expected.loc[float_expected.index[1], "Str2Col"] = 3.0
-        with tm.assert_produces_warning(
-            FutureWarning,
-            match="convert_float is deprecated",
-            raise_on_extra_warnings=False,
-        ):
-            # raise_on_extra_warnings because xlrd raises a PendingDeprecationWarning
-            # on database job Linux_py37_IO (ci/deps/actions-37-db.yaml)
-            # See GH#41176
-            actual = pd.read_excel(
-                basename + read_ext, sheet_name="Sheet1", convert_float=False
-            )
+        # raise_on_extra_warnings because xlrd raises a PendingDeprecationWarning
```

Comment:
Tangential to this PR but assuming we ultimately will be removing xlrd as well

https://github.com/pandas-dev/pandas/pull/35029

Reply:
Yep but did not want to mix too much stuff

Reply:
removed